src/vm_cpu.rs

```diff
@@ -96,6 +98,168 @@ pub struct VmCpuTrapState {
     pub htinst: u64,
 }
 
+#[derive(Default, Copy, Clone)]
+#[repr(C)]
+struct CounterState {
+    value: u64,
+    config_flags: u64,
+    event_type: u64,
+    event_data: u64,
+}
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+enum PmuCounterState {
+    NotConfigured,
+    Configured(CounterState),
+    Started(CounterState),
+    Stopped(CounterState),
```

Comment:
Aren't `Configured` and `Stopped` the same state?

Reply:
Possibly, but it seemed cleaner to model is as three distinct states.